### PR TITLE
NH-93485: export metrics via otlp

### PR DESCRIPTION
**Pull request by cleverchuk**
**Tl;dr**: OTLP metric export

**Context**:

Prefer exporting metrics via OTLP. Add some code to gate export via `apm-proto` only when OTLP is explicit disabled or collector is appoptics. Refactored and added some test to represent the current state.

**Test Plan**:
Test services data [0](https://my.na-01.st-ssp.solarwinds.com/205939959869206528/entities/services/e-1712644058766987264/overview?duration=21600), [1](https://my.na-01.st-ssp.solarwinds.com/205939959869206528/entities/services/e-1712643928659124224/overview?duration=21600), [2](https://my.na-01.st-ssp.solarwinds.com/205939959869206528/entities/services/e-1742334541200846848/logs?duration=21600) and [3](https://my.na-01.st-ssp.solarwinds.com/205939959869206528/entities/services/e-1777406072376840192/overview?duration=21600)

### CODE REVIEW

File: custom/src/main/java/com/solarwinds/opentelemetry/extensions/SolarwindsAgentListener.java [hunk 1]
```diff
@@ -152,10 +153,16 @@ protected MetricsMonitor buildMetricsMonitor() {
...
                                           .buildSpanMetricsCollector());
-                              metricsCollector.addCollector(
-                                  MetricsCategory.TRACING_REPORTER,
-                                  new TracingReporterMetricsCollector(
-                                      ReporterProvider.getEventReporter()));
+
+                              if (ConfigurationLoader.shouldUseOtlpForMetrics()) {
+                                metricsCollector.removeCollector(MetricsCategory.LAYER_COUNT);
+                              } else {
+                                metricsCollector.addCollector(
```
Comment (cleverchuk) [inline custom/src/main/java/com/solarwinds/opentelemetry/extensions/SolarwindsAgentListener.java:160]: Only add this if not `OTLP`. This is for exporting trace reporter stats metric which I'm assuming is safe to ignore.

**Review by cheempz — APPROVED** [review]
LGTM, thanks @cleverchuk! I'm not sure if your follow-on PR for OTLP export of traces already addressed the need to set resource attributes that correspond to apm-proto HostID also for metrics but makes sense to get this one in first :)
